File: Product-Community/product-community-frontend/src/Services/UserServices.js
import axios from "axios"
const API_URL = "http://localhost:7001/community";

class UserServices {

    addUser(user) {
        return axios.post(API_URL + "/signup", user);
    }

    userExists(email) {
        return axios.get(API_URL + "/user/" + email);
    }

    getUserAuth(email, password) {
        return axios.get(API_URL + "/login/" + email + "/" + password)
    }

    getUser(email) {
        return axios.get(API_URL + "/details/" + email)
    }

}

export default new UserServices();

File: Product-Community/ProductCommunity-Backend/src/main/java/com/nagarro/product/community/services/UserServices.java
package com.nagarro.product.community.services;

import com.nagarro.product.community.entity.User;

public interface UserServices {
	public void addUser(User user);
	public boolean userExists(String email);
	public User getUserById(String email);
	public User getUserAuthenticated(String userId,String password);
	
}
